<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
        "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>JavaScript unit test file</title>
  <meta http-equiv="content-type" content="text/html; charset=utf-8" />
  <script src="assets/prototype.js" type="text/javascript"></script>
  <script src="assets/unittest.js" type="text/javascript"></script>
  
  <script src="../src/ajax_mock.js" type="text/javascript"></script>
  
  <link rel="stylesheet" href="assets/unittest.css" type="text/css" />
  
</head>
<body>

<div id="content">

  <div id="header">
    <h1>JavaScript unit test file</h1>
    <p>
      This file tests <strong>ajax_mock.js</strong>.
    </p>
  </div>

  <!-- Log output (one per Runner, via {testLog: "testlog"} option)-->
  <div id="testlog"></div>
  
  <!-- Put sample/test html here -->
  <div id="sample"></div>
</div>

<script type="text/javascript">
// <![CDATA[

  new Test.Unit.Runner({
    setup: function() {
      this.working_url = "/working/url.json";
      this.nonworking_url = "/nonworking/url";
      this.completed = false;
      this.failed = false;
    },
    
    teardown: function() {
      Ajax.Request.clearMocks();
    },
    
    // Ajax call should complete
    testAjaxCallShouldComplete: function() { with(this) {
      self = this;
      Ajax.Request.setupMock(working_url, function(request, response) {
        response.responseJSON = "data";
        request.options.onComplete(response);
      });
      new Ajax.Request(working_url, {method:"get", 
        onComplete: function() {
          self.completed = true;
        }, 
        onFailure: function() {
          self.error("onFailure should not be called");
        }});
      wait(100, function() {
        assert(self.completed, 'onComplete should have been called');
      });
    }},

    // AjaxCallShouldFail
    testAjaxCallShouldFail: function() { with(this) {
      self = this;
      Ajax.Request.setupMock(nonworking_url, function(request, response) {
        request.options.onFailure(response);
      })
      new Ajax.Request(nonworking_url, {method:"get", 
        onComplete: function() {
          self.error("onComplete should not be called");
        }, 
        onFailure: function() {
          self.failed = true;
        }});
      wait(100, function() {
        assert(self.failed, 'onFailure should have been called');
      });
    }},

    
      
  }); 
// ]]>
</script>
</body>
</html>